From cde0ffd1614b66137268534fa0a282b89ae5ea46 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 5 Oct 2017 21:45:16 -0700
Subject: [PATCH] Monumenta - Removed test for monsters when sleeping in beds


diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
index f5d9b4abc..81747883a 100644
--- a/src/main/java/net/minecraft/server/EntityHuman.java
+++ b/src/main/java/net/minecraft/server/EntityHuman.java
@@ -1300,13 +1300,6 @@ public abstract class EntityHuman extends EntityLiving {
             }
 
             if (!this.u()) {
-                double d0 = 8.0D;
-                double d1 = 5.0D;
-                List<EntityMonster> list = this.world.a(EntityMonster.class, new AxisAlignedBB((double) blockposition.getX() - 8.0D, (double) blockposition.getY() - 5.0D, (double) blockposition.getZ() - 8.0D, (double) blockposition.getX() + 8.0D, (double) blockposition.getY() + 5.0D, (double) blockposition.getZ() + 8.0D), (Predicate) (new EntityHuman.c(this)));
-
-                if (!list.isEmpty()) {
-                    return EntityHuman.EnumBedResult.NOT_SAFE;
-                }
             }
         }
         return EntityHuman.EnumBedResult.OK;
-- 
2.17.1

