From 930947256d86233a080466f6cc562be52d48c48d Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Fri, 6 Sep 2019 20:36:55 -0400
Subject: [PATCH] Monumenta - DamageCause.CUSTOM support for class abilities

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 1f6cbd1d6..0ca554e05 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -726,6 +726,8 @@ public class CraftEventFactory {
                 } else if (damager.getBukkitEntity() instanceof Projectile) {
                     cause = DamageCause.PROJECTILE;
                 }
+            } else if ("custom".equals(source.translationIndex)) {
+                cause = DamageCause.CUSTOM;
             } else if ("thorns".equals(source.translationIndex)) {
                 cause = DamageCause.THORNS;
             }
-- 
2.17.1

