From 30f2491c1911aa2cda0cdc8571183fa5a0a818f9 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 28 Feb 2019 23:37:12 -0500
Subject: [PATCH] Monumenta - Increase kicked-for-flying timer from 4s to 6s

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 7929fcc80..e127bc5ba 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -151,7 +151,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
         ++this.e;
         this.processedMovePackets = this.receivedMovePackets;
         if (this.B) {
-            if (++this.C > 80) {
+            if (++this.C > 120) {
                 PlayerConnection.LOGGER.warn("{} was kicked for floating too long!", this.player.getDisplayName().getString());
                 this.disconnect(com.destroystokyo.paper.PaperConfig.flyingKickPlayerMessage); // Paper - use configurable kick message
                 return;
@@ -170,7 +170,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
             this.w = this.r.locY();
             this.x = this.r.locZ();
             if (this.D && this.player.getRootVehicle().getRidingPassenger() == this.player) {
-                if (++this.E > 80) {
+                if (++this.E > 120) {
                     PlayerConnection.LOGGER.warn("{} was kicked for floating a vehicle too long!", this.player.getDisplayName().getString());
                     this.disconnect(com.destroystokyo.paper.PaperConfig.flyingKickVehicleMessage); // Paper - use configurable kick message
                     return;
-- 
2.17.1

