From d4a7eaf20fba13bb51ed182dfda1db9fdee3eab3 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 6 Feb 2020 01:34:38 -0500
Subject: [PATCH] Monumenta - Refuse to start shard if scoreboard fails to load

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/WorldPersistentData.java b/src/main/java/net/minecraft/server/WorldPersistentData.java
index a2a25cf6a..9bfbe6c75 100644
--- a/src/main/java/net/minecraft/server/WorldPersistentData.java
+++ b/src/main/java/net/minecraft/server/WorldPersistentData.java
@@ -70,6 +70,7 @@ public class WorldPersistentData {
             }
         } catch (Exception exception) {
             WorldPersistentData.LOGGER.error("Error loading saved data: {}", s, exception);
+            System.exit(-9001);
         }
 
         return null;
-- 
2.17.1

