From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Fri, 9 Oct 2020 18:19:13 -0400
Subject: [PATCH] Monumenta - Remove wither nether star drop

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EntityWither.java b/src/main/java/net/minecraft/server/EntityWither.java
index 29bb74e51e73ab5b1ccc814b6599e0baf4d8094c..1549af64e6f5318ad6815c7e3e0e23e1801e1542 100644
--- a/src/main/java/net/minecraft/server/EntityWither.java
+++ b/src/main/java/net/minecraft/server/EntityWither.java
@@ -480,16 +480,6 @@ public class EntityWither extends EntityMonster implements IRangedEntity {
         }
     }
 
-    @Override
-    protected void dropDeathLoot(DamageSource damagesource, int i, boolean flag) {
-        super.dropDeathLoot(damagesource, i, flag);
-        EntityItem entityitem = this.a((IMaterial) Items.NETHER_STAR);
-
-        if (entityitem != null) {
-            entityitem.r();
-        }
-
-    }
 
     @Override
     public void checkDespawn() {
