From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Wed, 29 Apr 2020 21:17:15 +0000
Subject: [PATCH] Monumenta - Reset last player hurt time on taking any damage

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index 76185f042dd773f267dc7574116cec7bd8a28ca8..6f04f1c2cd897df17f6c55d8a31942a8b6bc019b 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -1180,6 +1180,10 @@ public abstract class EntityLiving extends Entity {
             this.ap = 0.0F;
             Entity entity1 = damagesource.getEntity();
 
+            if (this.killer != null) {
+                this.lastDamageByPlayerTime = 100;
+            }
+
             if (entity1 != null) {
                 if (entity1 instanceof EntityLiving) {
                     this.setLastDamager((EntityLiving) entity1);
