From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Sat, 23 Mar 2019 01:51:24 -0400
Subject: [PATCH] Monumenta - drowned no longer have biome restrictions

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EntityDrowned.java b/src/main/java/net/minecraft/server/EntityDrowned.java
index fb26cd58ed2a02926c2cd44fb0fc4029568e1993..6f35a4dd91403667f7350ae6569b9f6fd61dfd9a 100644
--- a/src/main/java/net/minecraft/server/EntityDrowned.java
+++ b/src/main/java/net/minecraft/server/EntityDrowned.java
@@ -50,7 +50,7 @@ public class EntityDrowned extends EntityZombie implements IRangedEntity {
         Optional<ResourceKey<BiomeBase>> optional = worldaccess.i(blockposition);
         boolean flag = worldaccess.getDifficulty() != EnumDifficulty.PEACEFUL && a(worldaccess, blockposition, random) && (enummobspawn == EnumMobSpawn.SPAWNER || worldaccess.getFluid(blockposition).a((Tag) TagsFluid.WATER));
 
-        return !Objects.equals(optional, Optional.of(Biomes.RIVER)) && !Objects.equals(optional, Optional.of(Biomes.FROZEN_RIVER)) ? random.nextInt(40) == 0 && a((GeneratorAccess) worldaccess, blockposition) && flag : random.nextInt(15) == 0 && flag;
+        return (enummobspawn == EnumMobSpawn.SPAWNER || random.nextInt(15) == 0) && flag;
     }
 
     private static boolean a(GeneratorAccess generatoraccess, BlockPosition blockposition) {
