From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Sun, 30 Aug 2020 15:49:58 -0400
Subject: [PATCH] Monumenta - Allow wolves to spawn from spawners on any block
 when not in peaceful

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
index dcbb34313fedb21e180a0b76610a787e6419d404..994b37a8d6c4bcb391dc1300a8b0d2195e0e9db3 100644
--- a/src/main/java/net/minecraft/server/EntityWolf.java
+++ b/src/main/java/net/minecraft/server/EntityWolf.java
@@ -33,6 +33,19 @@ public class EntityWolf extends EntityTameableAnimal implements IEntityAngerable
         this.setTamed(false);
     }
 
+    @Override
+    public boolean a(GeneratorAccess generatoraccess, EnumMobSpawn enummobspawn) {
+        if (generatoraccess.getDifficulty() == EnumDifficulty.PEACEFUL) {
+            return false;
+        } else if (enummobspawn == EnumMobSpawn.SPAWNER) {
+            // Spawner, allow spawning on any non-air block regardless of light
+            return !generatoraccess.getType(this.getChunkCoordinates().down()).isAir();
+        } else {
+            // Non-spawner, use default spawning rules
+            return super.a(generatoraccess, enummobspawn);
+        }
+    }
+
     @Override
     protected void initPathfinder() {
         this.goalSelector.a(1, new PathfinderGoalFloat(this));
