From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Tue, 31 Oct 2017 17:44:53 -0700
Subject: [PATCH] Monumenta - Charged creepers no longer cause mob head drops


diff --git a/src/main/java/net/minecraft/server/EntityCreeper.java b/src/main/java/net/minecraft/server/EntityCreeper.java
index 79ef955070b2982be79cc58e40093624bd088ff0..d18c9dc5f5b7bbd24ad7341895eeec1b6df5f91f 100644
--- a/src/main/java/net/minecraft/server/EntityCreeper.java
+++ b/src/main/java/net/minecraft/server/EntityCreeper.java
@@ -269,7 +269,7 @@ public class EntityCreeper extends EntityMonster {
     }
 
     public boolean canCauseHeadDrop() {
-        return this.isPowered() && this.bs < 1;
+        return false;
     }
 
     public void setCausedHeadDrop() {
