From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Fri, 9 Oct 2020 18:19:13 -0400
Subject: [PATCH] Monumenta - Remove wither nether star drop

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EntityWither.java b/src/main/java/net/minecraft/server/EntityWither.java
index 53482a420c356fd50e5ab6dd729a271d6594dac7..c33b45114752da73ab1301f9185a459138a0363e 100644
--- a/src/main/java/net/minecraft/server/EntityWither.java
+++ b/src/main/java/net/minecraft/server/EntityWither.java
@@ -480,16 +480,6 @@ public class EntityWither extends EntityMonster implements IRangedEntity {
         }
     }
 
-    @Override
-    protected void dropDeathLoot(DamageSource damagesource, int i, boolean flag) {
-        super.dropDeathLoot(damagesource, i, flag);
-        EntityItem entityitem = this.a((IMaterial) Items.NETHER_STAR);
-
-        if (entityitem != null) {
-            entityitem.r();
-        }
-
-    }
 
     @Override
     public void checkDespawn() {
