From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 16 Apr 2020 19:50:43 -0400
Subject: [PATCH] Monumenta - Max level for curse of vanishing is 2

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EnchantmentVanishing.java b/src/main/java/net/minecraft/server/EnchantmentVanishing.java
index e4ffab92483d50d83b186d918487ded76c97b2d8..452ae0b341c1c4864f7a681c39348643acb6f343 100644
--- a/src/main/java/net/minecraft/server/EnchantmentVanishing.java
+++ b/src/main/java/net/minecraft/server/EnchantmentVanishing.java
@@ -18,7 +18,7 @@ public class EnchantmentVanishing extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 1;
+        return 2;
     }
 
     @Override
