From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Mon, 5 Apr 2021 14:04:44 -0400
Subject: [PATCH] Monumenta - Move spawnpoint set for sleeping in bed after
 event

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
index de0efe0b0fedbb2665034dff9ac593042c8ff2dd..9e948b6321f4bc48cc3915f72aeca692739bc0cf 100644
--- a/src/main/java/net/minecraft/server/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/EntityPlayer.java
@@ -1097,13 +1097,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
             } else if (this.b(blockposition, enumdirection)) {
                 return Either.left(EntityHuman.EnumBedResult.OBSTRUCTED);
             } else {
-                this.setRespawnPosition(this.world.getDimensionKey(), blockposition, this.yaw, false, true);
-                if (this.world.isDay()) {
-                    return Either.left(EntityHuman.EnumBedResult.NOT_POSSIBLE_NOW);
-                } else {
-
-                    return Either.right(Unit.INSTANCE);
-                }
+                return Either.right(Unit.INSTANCE);
             }
         } else {
             return Either.left(EntityHuman.EnumBedResult.OTHER_PROBLEM);
@@ -1128,6 +1122,8 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
             return bedResult;
         }
 
+        this.setRespawnPosition(this.world.getDimensionKey(), blockposition, this.yaw, false, true);
+
         {
             {
                 {
