From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 4 Mar 2021 16:02:56 -0500
Subject: [PATCH] Monumenta - Decrease severity of tile entity fix to warning

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/server/level/WorldServer.java b/src/main/java/net/minecraft/server/level/WorldServer.java
index 7e032845fe9a46d33236444d62405a41e29daa17..b0ab8c001dce7b9b9db16165db5d805375d20ee2 100644
--- a/src/main/java/net/minecraft/server/level/WorldServer.java
+++ b/src/main/java/net/minecraft/server/level/WorldServer.java
@@ -399,7 +399,7 @@ public class WorldServer extends World implements GeneratorAccessSeed {
     }
 
     private TileEntity fixTileEntity(BlockPosition pos, Block type, TileEntity found) {
-        this.getServer().getLogger().log(Level.SEVERE, "Block at {0}, {1}, {2} is {3} but has {4}" + ". "
+        this.getServer().getLogger().log(Level.WARNING, "Block at {0}, {1}, {2} is {3} but has {4}" + ". "
                 + "Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.", new Object[]{pos.getX(), pos.getY(), pos.getZ(), type, found});
 
         if (type instanceof ITileEntity) {
